homeassistant/components/broadlink/config_flow.py

```diff
@@ -126,6 +137,89 @@ async def async_step_user(self, user_input=None):
             errors=errors,
         )
 
+    async def async_step_discover(self, user_input=None):
+        """Discover devices on the given networks.
+
+        If the host is empty or 255.255.255.255, discover devices on all
+        available networks.
+        """
+        host = user_input.get(CONF_HOST)
+        timeout = user_input.get(CONF_TIMEOUT)
+        errors = {}
+
+        if not host or host == "255.255.255.255":
+            nics = await self.hass.async_add_executor_job(psutil.net_if_addrs)
+            broadcast_addrs = get_broadcast_addrs(nics)
+
+        elif is_broadcast_addr(host):
+            broadcast_addrs = [host]
+
+        else:
+            user_input[CONF_TIMEOUT] = self.data.pop(CONF_TIMEOUT, DEFAULT_TIMEOUT)
+            return await self.async_step_user(user_input=user_input)
+
+        devices = []
+        already_configured = self._async_current_ids(False)
+        in_progress = [
+            progress["context"].get("unique_id")
+            for progress in self._async_in_progress()
+        ]
+
+        for addr in broadcast_addrs:
+            discover = partial(blk.discover, discover_ip_address=addr, timeout=timeout)
+            try:
+                new_devices = await self.hass.async_add_executor_job(discover)
+
+            except OSError as err:
+                if err.errno == errno.ENETUNREACH:
+                    errors["base"] = "cannot_connect"
+                    err_msg = str(err)
+                else:
+                    errors["base"] = "unknown"
+                    err_msg = str(err)
+
+            else:
+                new_devices = [
+                    device
+                    for device in new_devices
+                    if device.mac.hex() not in already_configured
+                    and device.mac.hex() not in in_progress
+                ]
+                devices.extend(new_devices)
+
+        if not devices:
+            if not errors:
+                errors["base"] = "no_devices_found"
+                err_msg = "No devices found"
+
+            _LOGGER.error("Failed to discover devices: %s", err_msg)
+            return self.async_abort(reason=errors["base"])
+
+        if errors:
+            _LOGGER.debug("Error during device discovery: %s", err_msg)
+
+        if len(devices) == 1:
```

Comment:
What is devices > 1 ?

Reply:
We show [this form](https://github.com/home-assistant/core/blob/fae4997e27919650e9dfba75dc7ab91f27df7100/homeassistant/components/broadlink/config_flow.py#L214-L221) and the user needs to select a device to configure.

Reply:
I get it now, you must must run the config_flow from the start for each device 👍